File: tools/agen/internal/actions/actions.go
package actions

import (
"os"
"path/filepath"
)

type Action struct {
ID    string
Label string
Cmd   string
}

func DetectActions(root string) ([]Action, error) {
var out []Action
if _, err := os.Stat(filepath.Join(root, "go.mod")); err == nil {
out = append(out, Action{ID: "go", Label: "go test", Cmd: "go test ./..."})
}
if _, err := os.Stat(filepath.Join(root, "package.json")); err == nil {
out = append(out, Action{ID: "npm", Label: "npm ci && npm test", Cmd: "npm ci && npm test"})
}
return out, nil
}

File: tools/agent-cli/lib/actions.js
const fs = require('fs');
const path = require('path');

function detectActions(root) {
  const actions = [];
  if (fs.existsSync(path.join(root, 'package.json'))) {
    actions.push({ id: 'npm', label: 'npm install & test', cmd: 'npm ci && npm test' });
  }
  if (fs.existsSync(path.join(root, 'pyproject.toml'))) {
    actions.push({ id: 'python', label: 'python -m pip install -r requirements.txt', cmd: 'python -m pip install -r requirements.txt' });
  }
  return actions;
}

module.exports = { detectActions };
